hs_core/tasks.py

```diff
@@ -203,6 +207,37 @@ def add_zip_file_contents_to_resource(pk, zip_file_path):
         os.unlink(zip_file_path)
 
 
+@shared_task
+def delete_zip(resource_id, zip_path):
+    from hs_core.hydroshare.utils import get_resource_by_shortkey
+    res = get_resource_by_shortkey(resource_id)
+    istorage = res.get_irods_storage()
+
+    irods_prefix = "/" + settings.IRODS_ZONE + "/home/" + settings.IRODS_USERNAME
+    full_zip_path = '{irods}/{path}'.format(irods=irods_prefix, path=zip_path)
+    istorage.delete(full_zip_path)
+

```

Comment:
@sblack-usu irods_prefix you set here does not take iRODS federation into account, replace it with ```res.root_path)``` which include resource_id in the root_path, so you may have to strip it out if input parameter zip_path also includes resource_id just to make sure resource_id is not duplicated in the full_zip_path.

Reply:
This was intentional, I'm storing all zips in the same place, for easy cleanup.  Each zip output path includes a random 32 bit string hash to avoid path clashes.  Is there anything wrong with this approach?

Reply:
Agreed that approach is better, but your current zip creation code does not do that. See my other comment for details.